errors.go

```diff
@@ -215,6 +235,36 @@ func (errorType ErrorType) UserErrorf(msg string, args ...interface{}) error {
 	}
 }
 
+// AddDetails adds a details element to an error
+func (errorType ErrorType) AddDetails(err error, details interface{}) error {
+	if err == nil {
+		return errorType.Details(details)
+	}
+
+	c := new(customError)
+	c.error = errors.WithStack(err)
+	c.userMessage = GetUserMessage(err)
+
+	c.details = append(GetDetails(err), details)
```

Comment:
at the beginning of this func add
```
if details == nil {
return err
}
```

otherwise, you might add `nil` objects to the details slice

Reply:
Makes sense. Done.